File: Java/src/main/java/domain/Seat.java
package domain;

public class Seat {
    public final String coach;
    public final int seatNumber;
    public final boolean isAvailable;

    public Seat(String coach, int seatNumber, String bookingReference) {
        this.coach = coach;
        this.seatNumber = seatNumber;
        this.isAvailable = "".equals(bookingReference);
    }
}

File: csharp/src/Domain/Seat.cs
namespace KataTrainReservation
{

        public struct Seat
        {
            public Seat(SeatId id, bool isAvailable)
            {
                Id = id;
                IsAvailable = isAvailable;
            }

            public SeatId Id { get; }
            public bool IsAvailable { get; }
        }
    }